src/main/java/com/alysaa/geyseradmintools/listeners/AdminToolOnDeath.java

```diff
@@ -9,10 +9,13 @@
 import java.util.List;
 
 public class AdminToolOnDeath implements Listener {
+
+    private static final ItemStack starTool = ItemStackFactory.getStarTool();
+
     @EventHandler
     public void onPlayerDeath(PlayerDeathEvent e) {
-        List<ItemStack> list = e.getDrops();
-        list.removeIf(item -> item.getType() == Material.NETHER_STAR);
+        // Destroy the star tool if a player dies with it
+        List<ItemStack> deathItems = e.getDrops();
+        deathItems.removeIf(itemStack -> itemStack.equals(starTool));
     }
```

Comment:
I don't think this applies if keep inventory is enabled, however, we check if the player already has the star tool item in the respawn listener so its no biggue